app/controllers/audits_controller.rb

```diff
@@ -0,0 +1,113 @@
+class AuditsController < ApplicationController
+  before_action :check_if_organization_admin
+  before_action :set_audit, only: %i(show edit update destroy finalize)
+
+  def index
+    @selected_location = filter_params[:at_location]
+    @audits = current_organization.audits.filter(filter_params)
+    @storage_locations = Audit.storage_locations_audited_for(current_organization).uniq
+  end
+
+  def show
+    @inventory_items = @audit.storage_location.inventory_items
+  end
+
+  def edit
+    redirect_to audits_path unless @audit&.in_progress?
+    @storage_locations = current_organization.storage_locations
+    @items = current_organization.items.alphabetized
+    @audit.line_items.build if @audit.line_items.empty?
+  end
+
+  def finalize
+    adjustment = Adjustment.new(organization_id: @audit.organization_id, storage_location_id: @audit.storage_location_id, comment: 'Created Automatically through the Auditing Process')
+    adjustment.save
+    @audit.update(adjustment_id: adjustment.id)
+    inventory_items = @audit.storage_location.inventory_items
+    inventory_items.each do |inventory_item|
+      line_item = @audit.line_items.find_by(item: inventory_item.item)
+      if line_item.nil?
+        @audit.adjustment.line_items.create(item_id: inventory_item.item.id, quantity: -inventory_item.quantity)
+      elsif line_item.quantity != inventory_item.quantity
+        @audit.adjustment.line_items.create(item_id: inventory_item.item.id, quantity: line_item.quantity - inventory_item.quantity)
+      end
+    end
+    @audit.storage_location.adjust!(@audit.adjustment)
+    @audit.finalized!
+    redirect_to audit_path(@audit), notice: "Audit is Finalized"
+  end
+
+  def update
+    @audit.line_items.destroy_all
```

Comment:
Can you clarify why you're doing this here? I'm not sure I understand the purpose.

Also -- What happens if the user attempts to `:update` a finalized `Audit`?

Reply:
OK -- this seems like it shouldn't happen, but it sounds like the issue may be elsewhere and would be out of scope for this feature. :+1: